Source/objects.cpp

```diff
@@ -1075,7 +1075,7 @@ void AddSarc(int i)
 	Objects[i]._oVar1 = GenerateRnd(10);
 	Objects[i]._oRndSeed = AdvanceRndSeed();
 	if (Objects[i]._oVar1 >= 8)
-		Objects[i]._oVar2 = PreSpawnSkeleton();
+		Objects[i]._oVar2 = PreSpawnSkeleton()->getId();

```

Comment:
previously if PreSpawnSkeleton() failed it would set ovar2 to -1, now it would result in UB, so we need to check if PreSpawnSkeleton() was successful.